CODE-OF-CONDUCT.md

```diff
@@ -0,0 +1,132 @@
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the
+  overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders (including maintainers, Working Group leads, the Technical
+Oversight Committee and the Steering Committee) are responsible for clarifying
+and enforcing our standards of acceptable behavior and will take appropriate
+and fair corrective action in response to any behavior that they deem
+inappropriate, threatening, offensive, or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the Istio Steering Committee at steering@discuss.istio.io.
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
+
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
+
+**Community Impact**: A violation through a single incident or series
+of actions.
+
+**Consequence**: A warning with consequences for continued behavior. No
```

Comment:
sometimes things get a little heated on slack or the like, even among good friends that have been working on this project for years. I'd hate to see this warning escalation occur for these long term relationships - where someone is just having a bad day and may lose their composure, yet the individuals may get along just fine.

I think what appears to be missing from this is who is ultimately responsible for enforcing this escalation? I get the steering committee can receive a report. But what about the scenario where the code of conduct was violated - in a heated moment - not out of malice and someone other than the parties that are interacting report a violation.

In this scenario, I personally believe the person wronged should have some say in the consequence, as perhaps they don't feel all that wronged, but other people feel that person was wronged.

Having seen where these code of conduct violations can lead in a community of brilliant engineers, I'd encourage some sense of judgment from the individual wronged, not that "mob mentality" rule any decision making.

Cheers,
-steve

Reply:
Steve, I don't think a spirited/heated discussion in my view implies a code of conduct violation unless the conversation went beyond the technical arguments. If you're suggesting that Steering Committee members should talk with both the parties before taking any decisions, I totally agree with that.

Reply:
The ultimate enforcement of the code of conduct is defined in the Istio Steering charter.

Reply:
@nrjpoddar yes this was my long-winded form of suggestion 💯

Reply:
@dcberg I was probably a little unclear. After re-reading what I wrote, @nrjpoddar teased out what I was after.

Reply:
I think we're agreed here: part of the overall judgement process involves the CoC interviewing the parties involved.

@ssuchter and @knrc we may want to document the process we covered in CoCC training - I think this'll help make folks looking to make reports feel more comfortable too.

Reply:
To close this out, I'll follow on with another PR next week that adds a second document here outlining the process for handling a CoC complain based on the CoCC workshop we did earlier this year.